libs/community/langchain_community/vectorstores/sqlserver.py

```diff
@@ -171,7 +177,9 @@ def _insert_embeddings(
             metadatas = [{} for _ in texts]
 
         if ids is None:
-            ids = [metadata.pop("id", uuid.uuid4()) for metadata in metadatas]
+            # Copy data from metadatas so data is not updated in-place.
+            metadata_to_format = copy.deepcopy(metadatas)
+            ids = [metadata.pop("id", uuid.uuid4()) for metadata in metadata_to_format]
```

Comment:
Without doing a copy of the original dictionary, the `pop()` action is done in-place on the dict and user data is being formatted.
e.g
text = ["weekday"]
metadata = [{"id":"1", "day":"Monday"}]
store.add_texts(text, metadata)

At the end of the script above, the value of metadata would be [{"day":"Monday"}].

One thing we can do to avoid doing a copy of the dictionary is to not modify the dict in the first place. In this case, the metadata column in the vector store will contain metadata that has `id`.

e.g.
**custom_id      id      metadata                                                     text                                                embedding**
0FF18BEE ,   9	 ,  {**"id": 9,** "summary": "Taste is neutral"} ,	The taste is like a regular cracker ,	0xA90180010000000000

Reply:
What is the desired end result of the metadata column: should it contain the id or not?  does it have to follow a specific format? what does PostgreSQL do?

Reply:
PostgreSQL leaves metadata the way it was provided by user. So, if it contains ID, it reflected in the metadata column in the DB.